hypothesis/h-periodic
# Upgrade to Python 3.8.9

For: https://github.com/hypothesis/h-periodic/issues/72

This PR:

* Upgrades the Python version to 3.8
* Upgrades the docker file to match
* Makes the GHA scripting reactive to the version listed in the `.python-version` file
* Recompiled the requirements files (after deleting those for format and lint, which had 3.7 incompatible libraries)

## Testing notes

 * In `checkmate` and `h` run: `make services dev`
 * Wait until any old queued scheduled tasks have run (they can build up)
 * Run `make dev`
 * You should see periodic tasks being kicked off. This is easy in `checkmate` which gets to work fast

You can also check the docker build with:

`make docker`
